core/templates/pages/exploration-player-page/services/question-player-engine.service.ts

```diff
@@ -39,14 +39,14 @@ import { FocusManagerService } from 'services/stateful/focus-manager.service';
   providedIn: 'root'
 })
 export class QuestionPlayerEngineService {
-  private explorationId: string = null;
-  private questionPlayerMode: boolean = null;
-  private version: number = null;
+  explorationId: string = null;
```

Comment:
Removed private variables since it is hard to test them without getting lint errors as we are injecting service into the Testbed, but not initializing a class itself.
So they are not available in the spec file.

I can think of fixing this issue using three methods
1. Bypassing lints in spec file https://stackoverflow.com/questions/55336796/access-a-private-variable-in-spec-ts-angular6
2. Removing private variables
3. adding setters and getters for each variable (I think, only for the sake of tests adding these are redundant)

Reply:
If variables are private then you shouldn't be accessing them directly in the spec file. Similar to backend tests, what we typically do is to test a public function which surfaces the variable. Otherwise, your tests should not be checking the values of those variables because they are considered "internal state". 

Remember, tests should test the public interface. It is important that they do not test the implementation. If the implementation changes but the behaviour is still correct then your tests should still pass with no issues. This is a really important principle to bear in mind when writing tests, otherwise you will have tests that are overfitted to the specific implementation that is in the code.

Reply:
done!